crates/iceberg-catalog/src/service/token_verification.rs

```diff
@@ -59,7 +78,8 @@ pub(crate) async fn auth_middleware_fn(
                 tracing::debug!("Failed to verify token: {:?}", err);
                 return IcebergErrorResponse::from(err).into_response();
             }
-        }
+        };
+        metadata.openid_config = Some(verifier.config.clone());
```

Comment:
entra id doesn't have a token info endpoint, there's only userinfo but userinfo cannot be called with the provided token, we either need service creds or do some oidc flow which results in a id token